sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/expressions.scala

```diff
@@ -416,6 +416,22 @@ object SimplifyConditionals extends Rule[LogicalPlan] with PredicateHelper {
         // these branches can be pruned away
         val (h, t) = branches.span(_._1 != TrueLiteral)
         CaseWhen( h :+ t.head, None)
+
+      case e @ CaseWhen(branches, Some(elseValue)) if {
+        // With previous rules, it's guaranteed that there must be one branch.
```

Comment:
Is this comment correct?

Reply:
You're right. I removed the comment. Thanks.